pkg/diagnostics/grpc_tracing.go

```diff
@@ -122,6 +122,17 @@ func SpanContextFromIncomingGRPCMetadata(ctx context.Context) (trace.SpanContext
 	if len(traceContext) > 0 {
 		traceContextBinary := []byte(traceContext[0])
 		sc, ok = propagation.FromBinary(traceContextBinary)
+	} else {
+		// add workaround to fallback on checking traceparent header
```

Comment:
I would add TODO: comment to recognize the code later.

Reply:
done.